### PR TITLE
Add information about the events of the accordion widget

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request adds information about **accordion** widget events.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_accordion.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_accordion.html
- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_accordion.html

## Links to Magento source code

Magento 2.3:

- https://github.com/magento/magento2/blob/2.3/lib/web/mage/accordion.js#L12
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/tabs.js#L159
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/collapsible.js#L444
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/collapsible.js#L481
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/collapsible.js#L519

Magento 2.2:

- https://github.com/magento/magento2/blob/2.2/lib/web/mage/accordion.js#L12
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/tabs.js#L159
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/collapsible.js#L438
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/collapsible.js#L476
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/collapsible.js#L514

Magento 2.1:

- https://github.com/magento/magento2/blob/2.1/lib/web/mage/accordion.js#L11
- https://github.com/magento/magento2/blob/2.1/lib/web/mage/tabs.js#L158
- https://github.com/magento/magento2/blob/2.1/lib/web/mage/collapsible.js#L357
- https://github.com/magento/magento2/blob/2.1/lib/web/mage/collapsible.js#L380
- https://github.com/magento/magento2/blob/2.1/lib/web/mage/collapsible.js#L414

whatsnew
Added information about [accordion widget events](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_accordion.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md [hunk 1]
```diff
@@ -179,6 +179,44 @@ $( "#element" ).accordion( "deactivate", 1 );
 $( "#element" ).accordion( "deactivate", [0,1]);
 ```
 
+## Events {#accordion_events}
+
+The accordion widget is subscribed to the same events as the [Magento Tabs widget] widget:
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md]: Please, use the link markup for `[Magento Tabs widget]`. So we have [Tabs Widget](https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_tabs.html)

**Comment by serhiyzhovnir** [comment]
Hi @rogyar,
We have fixed the link for Tabs widget.
Could you, please, check it again?
Thank you!

File: guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md [hunk 2]
```diff
@@ -179,6 +179,44 @@ $( "#element" ).accordion( "deactivate", 1 );
 $( "#element" ).accordion( "deactivate", [0,1]);
 ```
 
+## Events {#accordion_events}
+
+The accordion widget is subscribed to the same events as the [Tabs widget]({{page.baseurl}}/javascript-dev-guide/widgets/widget_tabs.html):
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md]:
```suggestion
The accordion widget subscribes to the same events as the [Tabs widget]({{page.baseurl}}/javascript-dev-guide/widgets/widget_tabs.html):
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md [hunk 3]
```diff
@@ -179,6 +179,44 @@ $( "#element" ).accordion( "deactivate", 1 );
...
+
+-   [beforeOpen callback](#accordion_beforeOpen)
+-   [dimensionsChanged](#accordion_dimensionsChanged)
+
+### `beforeOpen callback` {#accordion_beforeOpen}
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md]: It's not necessary to add header ID overrides. Jekyll automatically generates header IDs at build time.

File: guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md [hunk 4]
```diff
@@ -179,6 +179,44 @@ $( "#element" ).accordion( "deactivate", 1 );
...
+
+Example of adding a callback to `beforeOpen` events:
+
+```javascript
+$("#element").on("beforeOpen", function () {
+    // do something before opening the content
+});
+```
+
+### `dimensionsChanged` {#accordion_dimensionsChanged}
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md]: It's not necessary to add header ID overrides. Jekyll automatically generates header IDs at build time.

**Comment by serhiyzhovnir** [comment]
@jeff-matthews 
I have applied the required changes and removed the header IDs for events.
Thank you!

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Build timed out again, which is why tests failed.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests failed. Here's the log:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 


Ran on 4115 files!


- linking to internal hash #accordion_beforeOpen that does not exist
  *  _site/guides/v2.1/javascript-dev-guide/widgets/widget_accordion.html (line 392)
     <a href="#accordion_beforeOpen">beforeOpen callback</a>
  *  _site/guides/v2.2/javascript-dev-guide/widgets/widget_accordion.html (line 392)
     <a href="#accordion_beforeOpen">beforeOpen callback</a>
  *  _site/guides/v2.3/javascript-dev-guide/widgets/widget_accordion.html (line 392)
     <a href="#accordion_beforeOpen">beforeOpen callback</a>
- linking to internal hash #accordion_dimensionsChanged that does not exist
  *  _site/guides/v2.1/javascript-dev-guide/widgets/widget_accordion.html (line 393)
     <a href="#accordion_dimensionsChanged">dimensionsChanged</a>
  *  _site/guides/v2.2/javascript-dev-guide/widgets/widget_accordion.html (line 393)
     <a href="#accordion_dimensionsChanged">dimensionsChanged</a>
  *  _site/guides/v2.3/javascript-dev-guide/widgets/widget_accordion.html (line 393)
     <a href="#accordion_dimensionsChanged">dimensionsChanged</a>
rake aborted!
HTML-Proofer found 6 failures!```

File: guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md [hunk 5]
```diff
@@ -179,6 +179,44 @@ $( "#element" ).accordion( "deactivate", 1 );
 $( "#element" ).accordion( "deactivate", [0,1]);
 ```
 
+## Events {#accordion_events}
+
+The accordion widget subscribes to the same events as the [Tabs widget]({{page.baseurl}}/javascript-dev-guide/widgets/widget_tabs.html):
+
+-   [beforeOpen callback](#accordion_beforeOpen)
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_accordion.md]: @serhiyzhovnir,  I forgot to ask you to update these anchor links when I asked you to remove them from the headings below. That's what's causing the build to fail. You'll need to update these anchors.

- `[beforeOpen callback](#beforeopen-callback)`
- `[dimensionsChanged](#dimensionschanged)`

**Comment by serhiyzhovnir** [comment]
Hi @jeff-matthews 
The anchors were updated.
Thank you!

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**